source/common/http/filter/fault_filter.cc

```diff
@@ -78,31 +90,70 @@ FilterHeadersStatus FaultFilter::decodeHeaders(HeaderMap& headers, bool) {
     return FilterHeadersStatus::Continue;
   }
 
-  if (config_->runtime().snapshot().featureEnabled("fault.http.delay.fixed_delay_percent",
-                                                   config_->delayPercent())) {
-    uint64_t duration_ms = config_->runtime().snapshot().getInteger(
-        "fault.http.delay.fixed_duration_ms", config_->delayDuration());
+  if (config_->matchDownstreamCluster()) {
```

Comment:
As per comment above I think this name confuses the point of the flag. Which is: get specific cluster settings.